yunwoo/baekjoon/정렬/10814-나이순-정렬.js

```diff
@@ -0,0 +1,33 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const userArr = input
+    .slice(1, input.length)
+    .map((item) => item.split(" "))
+    .sort((a, b) => +a[0] - +b[0]);
+  let result = "";
+
+  for (let i = 0; i < +input[0]; i++) {
+    for (let j = 0; j < userArr[i].length; j++) {
+      result += !j ? `${userArr[i][j]} ` : `${userArr[i][j]}`;
+    }
+    result += "\n";
+  }
+
+  console.log(result);
```

Comment:
```suggestion
const userArr = input
  .slice(1)
  .map((item) => item.split(" "))
  .sort((a, b) => +a[0] - +b[0]);
let result = "";

userArr.forEach(([age, name]) => {
  result += `${age} ${name}\n`;
});

console.log(result);
```
- 풀이 잘봤습니다~ 🙇‍♂️
- slice를 할 때 배열 끝까지 자른다면 2번째 인자는 필요 없습니다.
- 2중 for문을 한번 더 하실 필요없이 forEach 한번으로 끝낼 수 있습니다 ! 👍

Reply:
- slice 끝까지 자를때는 시작number만!! 기억하겠습니다 🙇‍♂️
- forEach를 고려하긴 했는데 불가능할거라 생각했는데 훨씬 좋은 방법이 있었군요..! 인자를 배열로 2개를 받으면 되는거였네요!!!
- 형변환은 전혀몰랐는데 감사합니다 🙇‍♂️👍

Reply:
- 넵 맞습니다🙆‍♂️  userArr을 콘솔로 찍어보면 `[ [ '20', 'Sunyoung' ], [ '21', 'Junkyu' ], [ '21', 'Dohyun' ] ]` 와 같은데 이를 userArr를 forEach로 순회하면서 현재 아이템을 구조분해할당으로 활용 한겁니다 👍

```js
userArr.forEach((item) => {
  console.log(item);
  // [ 20, 'Sunyoung' ]
  // [ 21, 'Junkyu' ]
  // [ 21, 'Dohyun' ]
});

// 위 item을 구조분해 할당으로 활용 

userArr.forEach(([age, name]) => {
  console.log(age, name);
  // 20 'Sunyoung'
  // 21 'Junkyu' 
  // 21 'Dohyun' 
});
```

Reply:
- 자세한 설명 감사합니다!!! 구조분해할당인가 ? 했었다가 `[age,name]  = item` 형식이어야 하지 않나? 싶었는데 이것도 구조분해 할당이 맞네요ㅎㅎ👍